### ISSUE TITLE
Memcache extension doesn't play nice with large MTU's?

### ISSUE THREAD

**Issue by slawomir-pryczek** [websupport-sk/pecl-memcache#8]
Hi, i come across a very weird issue when i SET (GET is not affected) a key with long value for memcached via memcache plugin and loopback interface. Basically requests on remote hosts were 50-100 times faster than on loopback.

After doing some tests, i narrowed the problem to large MTU of loopback interface. So as you can see setting 3x30kb of data takes 80ms when connecting to memcache using TCP on localhost. It takes <1 ms when memcache is on remote machine.

screencast.com/t/bja6rafah

To fix the issue, we can do following:
1. Set memcache.chunk_size to be larger than item size (which is probably pointless to use so large buffers).
OR
2. Lower the MTU on loopback to 1500 which is standard for other interfaces, as on picture below:

http://screencast.com/t/XGp2MtkbQIgn

The 65k MTU is probably there for a reason, so you think it could be some problem with memcache plugin's networking code?

```
I found some related info:
http://comments.gmane.org/gmane.comp.web.cache.memcached/1259
When the problem occur it's a whole lot slower, more like:
7168 bytes: 8680 set/s
8192 bytes: 25 set/s
```

UPDATE: Actually if you set MTU to
`ifconfig lo mtu 32820 up`
it's working ok with req time <1ms, after you increase just by 1 byte to 32821 - the requests start to take over 80ms.

Thanks,
Slawomir.

**Comment by slawomir-pryczek** [comment]
Attaching test code + images.






[mcbench.txt](https://github.com/websupport-sk/pecl-memcache/files/209288/mcbench.txt)
![2016-04-07_2322](https://cloud.githubusercontent.com/assets/7211190/14367970/240f109e-fd1b-11e5-9e9b-4745bd85e2c7.png)
![2016-04-07_2330](https://cloud.githubusercontent.com/assets/7211190/14367971/242ad1ee-fd1b-11e5-998f-ad5bd8814df4.png)

**Comment by thesource93** [comment]
Couldn't reproduce the problem on any recent version , we then tried older (2016) version of memcached daemon, the behaviour appeared. The bug is fixed in any somewhat recent memcached.

**Comment by Zaffy** [comment]
TL;DR;  debugged this, and there's nothing to do, it's fixed.

First I thought of Nagle's algorithm and tried setting TCP_NODELAY to see if it'd make any difference. Not at all.
I just couldn't reproduce it, even with small chunk sizes.

Then there's this old conversation:
https://web.archive.org/web/20130213153546/http://comments.gmane.org/gmane.comp.web.cache.memcached/1259

It seems like the issue was in memcache daemon itself - not playing nice with fragmented packets, not in this extension. I pulled an old version of memcached  from 2016. And now, I could reproduce this.

The up-to-date version works like expected.
